ext/standard/array.c

```diff
@@ -3801,29 +3801,37 @@ PHP_FUNCTION(array_change_keys)
         fci.no_separation = 0;
 
         if (str_key) {
-            ZVAL_STR(&params[0], zend_string_copy(str_key));
+            ZVAL_STR_COPY(&params[0], str_key);
```

Comment:
`ZVAL_STR_COPY()` is the correct macro to use here. It will likely do the same thing in practice, but has extra internal checks.